src/QsFmt/Formatter.Tests/Discoverer.fs

```diff
@@ -54,17 +64,44 @@ module internal Example =
 /// Marks a property of type <c>string * string</c> that should be run as an example test case, with the first item
 /// being the source code before and the second item being the source code after.
 /// </summary>
-type ExampleAttribute() =
+type FormatExampleAttribute() =
     inherit Attribute()
 
     /// A reason for skipping the test, if non-null. The test should not be skipped if null.
     member val Skip: string = null with get, set
 
+/// A test case containing an example of source code before and after updating.
+type UpdateExample =
```

Comment:
I'm really unhappy with having to duplicate the original 'Example' types to be able to determine which examples get run against Formatter.format versus Formatter.update. I'd really appreciate someone with better F# savvy to offer a better way of accomplishing this.

Reply:
You could add a parameter to the `ExampleAttribute` constructor, and a field to the `Example` record, that says whether it's a format or update example. It could be a type like `type ExampleKind = Format | Update`.

Reply:
I can't seem to pass in as an argument to the attribute the kind of example it is.
```
[<Example(ExampleKind.FormatExample)>]
let ``Indents function`` = ...
```
This just complains. This is an F# thing.

Reply:
You might need to use an enum. There are .NET restrictions that require attribute expressions to contain only compile-time constants, and I think .NET doesn't recognize an F# DU as being constant.

Reply:
Yah I don't think F# supports passing DU cases as arguments, at least not to attribute constructors. I can change it to be a Boolean flag, and that works, but that is pretty janky. To be honest, I think this whole setup with the attributes is a little overkill with the complexity, and I'd rather just work with a flat list of unit tests rather than one unit test that does everything. That would integrate better with visual studio's testing framework as well.

Reply:
> Yah I don't think F# supports passing DU cases as arguments, at least not to attribute constructors. I can change it to be a Boolean flag, and that works, but that is pretty janky.

An enum should work: `type ExampleKind = Format = 0 | Update = 1`.

> To be honest, I think this whole setup with the attributes is a little overkill with the complexity, and I'd rather just work with a flat list of unit tests rather than one unit test that does everything. That would integrate better with visual studio's testing framework as well.

Well, the individual examples should still show up as separate things in the test list, since it's using the normal Xunit Theory feature. The benefit of the attributes is that you can write out your tests declaratively as top-level values and avoid some boilerplate. It is a little janky so you could try something different.

Reply:
Another option is not to add anything to the attribute itself, but to replace the `string * string` pair with a richer type, like the `Example` record. You could define convenience functions to create common types of examples.

Reply:
Sorry, GitHub was not loading your responses. I'm just now seeing your suggestions for the .NET Enums.

Reply:
Using the enums worked for now. Thanks!